src/controllers/credentials/index.ts

```diff
@@ -0,0 +1,51 @@
+import express from 'express'
+import { Get, Produces, Query, Request, Route, Security, SuccessResponse } from 'tsoa'
+import { injectable } from 'tsyringe'
+
+import VeritableCloudagent, { Credential } from '../../models/veritableCloudagent.js'
+import CredentialListTemplates from '../../views/credentials/index.js'
+import { HTML, HTMLController } from '../HTMLController.js'
+
+@injectable()
+@Security('oauth2')
+@Route('/credentials')
+@Produces('text/html')
+export class CredentialsController extends HTMLController {
+  constructor(
+    private credentialsTemplates: CredentialListTemplates,
+    private cloudagent: VeritableCloudagent
+  ) {
+    super()
+  }
+
+  @SuccessResponse(200)
+  @Get('/')
+  public async listCredentials(@Request() req: express.Request, @Query() search: string = ''): Promise<HTML> {
+    const credentials: Credential[] = await this.cloudagent.getCredentials()
+    req.log.info('retrieved credentials from a cloudagent %j', credentials)
+
+    const combined = credentials
+      .map((cred: Credential) => {
+        if (cred.credentialAttributes) {
+          return {
+            ...cred,
+            updated_at: new Date(),
+            company_name: cred?.credentialAttributes[0].name === 'company_name' && cred.credentialAttributes[0].value,
+            company_number:
+              cred.credentialAttributes[1].name === 'company_number' && cred.credentialAttributes[1].value,
+          }
+        }
+      })
+      .filter((cred) =>
+        search !== '' ? cred?.company_name.toString().toLowerCase().includes(search.toLowerCase()) : true
+      )
```

Comment:
Couple of comments here. This assumes that we're looking at only this type of credential and doesn't check that this alligns with a connection in the ui database. Can I suggest two improvements:

1. Filter this list on credentials that are of type `COMPANY_DETAILS`. That way we're not misinterpretting other credentials accidentally if they happen to exist
2. Filter this list based on the other sides connection id. You can probably fetch all connections first and build a `Set` of the `agent_connection_id` values. Then check for membership of the set in a filter

Also a question, do we want to only filter credentials in certain states. That way we don't show things in flight? Do we know how to display each credential status?

Reply:
Good points and agent connection id is basically our source of truth
> Also a question, do we want to only filter credentials in certain states. That way we don't show things in flight? Do we know how to display each credential status?

indeed we do have quite a states of cred until we reach done. They change fairly quick so really I was able to observe a few states until it reached done state. In terms of filter/table/creds i got some notes: 
- filter only creds that I was 'issuer'
- not sure if makes sense, noticed that either holder or issuer, can't remember which one would store connectionIds: [<ids>] to pull connection/company so state of connection can be seen (e.g. verification process like verified_us...)
- render somewhere attributes like on mouse over and also manage them better, looks like some generic helpers for formatting/checking/etc

To be fair when it comes to table columns I think they could different depending on whats relevant to the end-user